www/js/goals.js

```diff
@@ -61,37 +81,340 @@ angular.module('emission.main.goals',['emission.services', 'ngSanitize'])
+			var members = response.data;
+			console.log(response.data);
+			members.forEach(function(user){
+				$scope.membersName.push(user.profile.name);
+			});
+		}, function(error){
+			console.log("Error when fetching members");
 		});
+	};
+
+	var bikeChallenge = function() {
+		var callOpts = {'method': 'GET', 'method_url': "/api/v3/challenges/8a8134d6-066d-424d-8f3d-0b559c2c1e78",
+							'method_args': null};
+
+		    	CommHelper.habiticaProxy(callOpts, function(response){
+					console.log("Sucessfully got bike challenge");
+					console.log(response);
+					$scope.challenges.push(response.data);
+				}, function(error){
+					console.log("Error when getting bike challenge");
+				});
+	};
+
+	var carpoolChallenge = function() {
+		var callOpts = {'method': 'GET', 'method_url': "/api/v3/challenges/d3e0ee13-8922-47ef-86a0-2c2f662585e1",
+							'method_args': null};
+
+		    	CommHelper.habiticaProxy(callOpts, function(response){
+					console.log("Sucessfully got carpool challenge");
+					console.log(response);
+					$scope.challenges.push(response.data);
+				}, function(error){
+					console.log("Error when getting carpool challenge");
+				});
+	};
+
+	var publicTransChallenge = function() {
+		var callOpts = {'method': 'GET', 'method_url': "/api/v3/challenges/581aea56-8f1f-42fa-ae1c-c6608bc780d5",
+							'method_args': null};
+
+		    	CommHelper.habiticaProxy(callOpts, function(response){
+					console.log("Sucessfully got public transport challenge");
+					console.log(response);
+					$scope.challenges.push(response.data);
+					console.log($scope.challenges);
+				}, function(error){
+					console.log("Error when getting public transport challenge");
+				});
+	};
+
+	var getChallenges = function(){
+  		bikeChallenge();
+		carpoolChallenge();
+		publicTransChallenge();
+  	};
+
+	$scope.joinChallenge = function(challengeId) {
+		var callOpts = {'method': 'POST', 'method_url': "/api/v3/challenges/"+challengeId+"/join",
+							'method_args': null};
+
+		    	CommHelper.habiticaProxy(callOpts, function(response){
+					console.log("Sucessfully joined the challenge");
+					getUserTask();
+					console.log(response);
+				}, function(error){
+					console.log("Error when joining the challenge");
+				});
+	};
+
+	//Tab switch
+	$scope.isActive = false;
+  	$scope.activeButton = function() {
+  		if($scope.isActiveP == true){
+  			$scope.partyButton();
+  		}
+    	$scope.isActive = !$scope.isActive;
+  	};
+
+  	$scope.isActiveP = false;
+  	$scope.partyButton = function() {
+  		if($scope.isActive == true){
+  			$scope.activeButton();
+  		}
+    	$scope.isActiveP = !$scope.isActiveP;
+  	};
+
+  	var refreshInfo = function(){
+		console.log("Refreshing information");
+		$scope.challenges=[];
+		getUserInfo();
+		getMembers();
+		getUserTask();
+		getChallenges();
+	};
+
+	refreshInfo();
+
+	$scope.refreshPage = function() {
+		console.log("Refreshing page");
+		refreshInfo();
+		$scope.$broadcast('scroll.refreshComplete');
```

Comment:
I think that you should only broadcast when all the refreshing is complete. Note that `refreshInfo()` is not synchronous